src/mmds/src/lib.rs

```diff
@@ -5,15 +5,24 @@
 extern crate lazy_static;
 extern crate serde_json;
 
+extern crate dumbo;
+extern crate logger;
 extern crate micro_http;
+extern crate snapshot;
+extern crate utils;
+extern crate versionize;
+extern crate versionize_derive;
 
 pub mod data_store;
+pub mod ns;
+pub mod persist;
 
 use serde_json::{Map, Value};
 use std::sync::{Arc, Mutex};
 
 use data_store::{Error as MmdsError, Mmds};
-use micro_http::{Body, Request, RequestError, Response, StatusCode, Version};
```

Comment:
My thought is that `build_response` function need to be implemented at the level of `mmds` crate. As you've provided right now, we would need to consume `build_response` from `dumbo` to create `micro-http::Response`s, which is not ergonomic when `dumbo::build_response` would need to be extended, to add other headers lets say, to the provided response. For example, if one needs to add an extra `micro-http::Header` to the response, it will need to call first `dumbo::build_response`, then take a mutable reference to it, and then add the header. Another option will be to adjust this `dumbo::build_response` function for multiple response building cases but I do not think it is in the scope of the `dumbo` crate. `micro-http::Response` already provides this API for building responses for multiple cases, so reinventing a build response API in `dumbo` does not worth it.  I suggest to make this response builder function implementation specific. What do you think?

Reply:
@iulianbarbu 
I agree with your thought. Indeed I implemented `build_response` function in `dumbo` crate only for using it in `dumbo::parse_request` function. If we implement `build_response` in `mmds` crate, how should we pass `mmds::build_response` to `dumbo::parse_request`?
I think the first way to do that is passing `mmds::build_response` through one of the argument like `callback` as I've already implemented. Do you have any other ideas?

Reply:
Passing the response through the callback is perfectly fine, as far as I can tell. I do not have any issue with it.